src/vs/editor/contrib/codeAction/browser/codeActionMenu.ts

```diff
@@ -101,7 +105,30 @@ export class CodeActionMenu extends Disposable {
 			domForShadowRoot: useShadowDOM ? this._editor.getDomNode()! : undefined,
 			getAnchor: () => anchor,
 			getActions: () => menuActions,
-			onHide: () => {
+			onHide: (didCancel) => {
+				const openedFromString = (options.fromLightbulb) ? CodeMenuOpenedFrom.Lightbulb : trigger.triggerAction;
+
+				type ApplyCodeActionEvent = {
+					codeActionFrom: CodeMenuOpenedFrom;
+					validCodeActions: any;
+					cancelled: boolean;
+				};
+
+				type ApplyCodeEventClassification = {
+					codeActionFrom: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'The kind of action used to opened the code action.' };
+					validCodeActions: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'The total number of valid actions that are highlighted and can be used.' };
+					cancelled: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'The indicator if the menu was selected or cancelled.' };
+					owner: 'mjbvz';
+					comment: 'Event used to gain insights into which code actions are being triggered';
+				};
+
+				this._telemetryService.publicLog2<ApplyCodeActionEvent, ApplyCodeEventClassification>('codeAction.applyCodeAction', {
+					codeActionFrom: <CodeMenuOpenedFrom>openedFromString,
```

Comment:
Left the casting for now - not sure how I could set defaults from the interface (since there were a LOT of times where CodeActionTrigger were being called)

Reply:
This is because `openedFromString` is `CodeMenuOpenedFrom | undefined`. The cast is hiding an error here because we are always expecting a value, not `undefined`